src/flint/fmpz.jl

```diff
@@ -1480,7 +1480,37 @@ remove(x::fmpz, y::Integer) = remove(x, fmpz(y))
 
 remove(x::Integer, y::fmpz) = remove(fmpz(x), y)
 
-remove(x::Integer, y::Integer) = remove(fmpz(x), fmpz(y))
+function remove(a::UInt, b::UInt)
+   b <= 1 && error("Factor <= 1")
+   a == 0 && error("Not yet implemented")
+   q = Ref(a)
```

Comment:
Does this modify its input? Is that what the interface for remove is supposed to do?

Reply:
How can you modify an immutable? @thofma

Reply:
How do you reach the unreachable?

Reply:
I guess so. But why the strange looking attempt to do so?

Reply:
Does not `n_remove2_precomp` have signature `(Ptr{UInt}, UInt, Float64)`?

Reply:
Sure, but I'm reacting to the fact that Ref wraps a, one of the inputs. Surely this is on the stack and therefore local and probably everything is ok, but it just seems weird. Go with it if that's what you want. It just looks slightly confusing.

Reply:
Feel free to rewrite it in a way that is less confusing.